jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpField.java

```diff
@@ -668,4 +668,47 @@ public long getLongValue()
             return _long;
         }
     }
+
+    static class MultiHttpField extends HttpField
+    {
+        private final List<String> _list;
+
+        public MultiHttpField(String name, List<String> list)
+        {
+            super(name, buildValue(list));
+            _list = list;
+        }
+
+        private static String buildValue(List<String> list)
+        {
+            StringBuilder builder = null;
+            for (String v : list)
+            {
+                if (StringUtil.isBlank(v))
+                    continue;
+                if (builder == null)
+                    builder = new StringBuilder(list.size() * v.length() * 2);
+                else
+                    builder.append(", ");
+                builder.append(v);
+            }
+
+            return builder == null ? null : builder.toString();
```

Comment:
In case of `List.of(" ", " ")`, we return `null`, but since fields with an empty value are allowed, should we instead return the empty string?

Reply:
I've been struggling with this.  If we allow blanks, then should the result be " , " or "" or null?
I think the simplest thing to do is to turn null entries into "" and then allow blank list entries as as well.  It means there can be list values like ", ,   ,   , " , which are invalid according to https://www.rfc-editor.org/rfc/rfc9110#section-5.6.1, but then we don't prevent invalid values from being set directly.   Also that suggests that blank list entries are OK, so long as there is at least one non blank, which is harder to enforce and explain.

Reply:
Ok I am confused.

I think we should discard `null` and blank strings, as the spec in 5.6.1.1 says that the sender must not generate empty elements.
But the tests explicitly test for values such as ", ,  ", which I don't think it's what we want.

Reply:
@sbordet Now I'm confused?  You said above that you wanted to convert the invalid list `List.of(" ", " ")` into the valid `""` empty value.  I could not find a sensible way of doing that generally.   If we just skip null/blank items, we can end up with the empty list, then it is unclear if that is a remove or an add of the empty value.

So instead I took the approach of what we do with other `add`/`put` methods, in that we do not enforce validity.   I think I explained that well enough above.

I think not enforcing validity is reasonable.   I could also live with throwing an IAE if we are given and invalid list.  I'm not so keen on trying to convert an invalid list into something valid.

Reply:
I have changed to throw IAE.